internal/uniqid/legacy.go

```diff
@@ -0,0 +1,50 @@
+package uniqid
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+)
+
+const legacyPersistDir = "activestate/persist"
+
+func moveLegacyFile(destination string) error {
+	legacyDir, err := legacyStorageDir()
+	if err != nil {
+		return errs.Wrap(err, "Could not get legacy storage directory")
+	}
+
+	// If the legacy file does not not exist there is nothing to move
+	if !fileExists(filepath.Join(legacyDir, fileName)) {
+		return nil
+	}
+
+	destinationDir := filepath.Dir(destination)
+	err = mkdir(destinationDir)
+	if err != nil {
+		return errs.Wrap(err, "Could not create new persist directory")
+	}
+
+	err = moveAllFiles(legacyDir, destinationDir)
+	if err != nil {
+		return errs.Wrap(err, "Could not move legacy uniqid file")
+	}
+
+	// The legacy directory is a sub directory, we want to remove the parent
+	err = os.RemoveAll(filepath.Dir(legacyDir))
```

Comment:
This should also just remove the one file, and the folder itself. If there are other files there then we don't "own" those so we should probably error out anyway.

Reply:
If we remove just the directory containing the uniqID file then the `activestate` directory will still exist at the root of `AppData`. I can add code to check that the `activestate` directory contains only one subdirectory and that directory only contains one file, but that feels like overkill considering we're the ones who own this directory and this is the only package that's writing to it.

I could also be misunderstanding what you're asking for 😛

Reply:
I would just do a remove of the uniqid file itself, followed by a removal of the dir. If either failed; ignore it. I see this as us doing minimal effort to retain the ID. If it failed because of permissions or something then we'll just end up with a new ID.